pkg/service/healthchecksvc/service.go

```diff
@@ -98,7 +100,20 @@ func (s *Service) Check(ctx context.Context) error {
 		// @TODO validate target schema against targetgroup
 		if deploymentItem.TargetGroupAssignment != nil {
 			// This needs to be replaced with an actual check
-			deploymentItem.TargetGroupAssignment.Valid = true
+
+			// need to do a structural comparison of the scheam for the provider in the describe against the schema stored on the target group
+
+			//lookup target group
+			targetGroup := storage.GetTargetGroup{ID: deploymentItem.TargetGroupAssignment.TargetGroupID}
+
+			_, err := s.DB.Query(ctx, &targetGroup)
+			if err != nil {
+				return err
+			}
+
+			valid := s.validateProviderSchema(targetGroup.Result.TargetSchema.Schema.AdditionalProperties, describeRes.Schema.Target.AdditionalProperties["Default"].Schema.AdditionalProperties)
```

Comment:
Can you please replace the variable assignment with a direct assignment to the struct?

```suggestion
			valid := s.validateProviderSchema(targetGroup.Result.TargetSchema.Schema.AdditionalProperties, describeRes.Schema.Target.AdditionalProperties["Default"].Schema.AdditionalProperties)
```